exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DrillSqlOperator.java

```diff
@@ -64,4 +108,47 @@ public boolean isDeterministic() {
   public List<DrillFuncHolder> getFunctions() {
     return functions;
   }
+
+  public static class DrillSqlOperatorBuilder {
+    private String name;
+    private final List<DrillFuncHolder> functions = Lists.newArrayList();
+    private int argCountMin = Integer.MAX_VALUE;
+    private int argCountMax = Integer.MIN_VALUE;
+    private boolean isDeterministic = true;
+
+    public DrillSqlOperatorBuilder setName(final String name) {
+      this.name = name;
+      return this;
+    }
+
+    public DrillSqlOperatorBuilder addFunctions(Collection<DrillFuncHolder> functions) {
+      this.functions.addAll(functions);
+      return this;
+    }
+
+    public DrillSqlOperatorBuilder setArgumentCount(final int argCountMin, final int argCountMax) {
+      this.argCountMin = Math.min(this.argCountMin, argCountMin);
+      this.argCountMax = Math.max(this.argCountMax, argCountMax);
+      return this;
+    }
+
+    public DrillSqlOperatorBuilder setDeterministic(boolean isDeterministic) {
+      if(this.isDeterministic) {
+        this.isDeterministic = isDeterministic;
+      }
+      return this;
+    }
+
+    public DrillSqlOperator build() {
+      return new DrillSqlOperator(
+          name,
+          functions,
+          argCountMin,
+          argCountMax,
+          isDeterministic,
+          TypeInferenceUtils.getDrillSqlReturnTypeInference(
+              name,
```

Comment:
Can we ensure that name and functions are always set, before call this constructor this builder? If not, will it hit problem?